src/daft-csv/src/metadata.rs

```diff
@@ -43,15 +71,42 @@ pub fn read_csv_schema(
     })
 }
 
+pub async fn read_csv_schema_bulk(
+    uris: &[&str],
+    parse_options: Option<CsvParseOptions>,
+    max_bytes: Option<usize>,
+    io_client: Arc<IOClient>,
+    io_stats: Option<IOStatsRef>,
+) -> DaftResult<Vec<(Schema, CsvReadStats)>> {
+    let handles_iter = uris.iter().map(|uri| {
+        let owned_string = uri.to_string();
+        let owned_client = io_client.clone();
+        let owned_io_stats = io_stats.clone();
+        let owned_parse_options = parse_options.clone();
+        tokio::spawn(async move {
+            read_csv_schema_single(
+                &owned_string,
+                owned_parse_options.unwrap_or_default(),
+                max_bytes,
+                owned_client,
+                owned_io_stats,
+            )
+            .await
+        })
+    });
+    let all_schemas = try_join_all(handles_iter)
```

Comment:
you may want buffer this instead

Reply:
Hmm I didn't think it was necessary since we're capping the bytes read at 1 MiB for now, but I suppose that this could get hairy when getting into the thousands of files range. 👍